src/pages/ProductCard/ProductCard.tsx

```diff
@@ -0,0 +1,77 @@
+import { useParams } from 'react-router';
+import React, { useEffect, useState } from 'react';
+import { getProducts } from '../../api/products';
+import { getPhoneById } from '../../api/phones';
+import { getTabletById } from '../../api/tablets';
+import { getAccessorieById } from '../../api/accessories';
+import { Item } from '../../types/Item';
+
+const defaultItem = {
+  id: '',
+  namespaceId: '',
+  name: '',
+  capacityAvailable: [],
+  capacity: '',
+  priceRegular: 0,
+  priceDiscount: 0,
+  colorsAvailable: [],
+  color: '',
+  images: [],
+  description: [],
+  screen: '',
+  resolution: '',
+  processor: '',
+  ram: '',
+  camera: '',
+  zoom: '',
+  cell: [],
+};
+
+export const ProductCard: React.FC = () => {
+  const { itemId } = useParams();
+  const [foundProduct, setFoundProduct] = useState<Item>(defaultItem);
+
+  useEffect(() => {
+    const fetchData = async () => {
+      try {
+        const products = await getProducts();
+        const product = products
+          .find(productToFind => productToFind.itemId === itemId) || null;
```

Comment:
How about passing a product type by passing a prop through routing? Depending on the path there will be different product type, so you don't have to send another GET request